app/models/concerns/tagable.rb

```diff
@@ -34,6 +34,13 @@ def tag(name_or_id)
                               tagable_id:     self.id)
   end
 
+  def tag_without_callbacks(name_or_id)
+    Tagging.skip_callback(:save, :after, :update_tagable_timestamp)
```

Comment:
this is cool. i sort of intuit why we do this but could perhaps use some extra explanation. my guess: we want to not update an entity's timestamp to the time we restored it when we re-tag it as a side-effect of restoring it from deletion. is that correct? are there other associations for which restoring an entity requires safe-guards like this? if not, why are taggings special?